src/Eagle.ts

```diff
@@ -714,30 +699,48 @@ export class Eagle {
         if( selection instanceof Edge){
             GraphRenderer.setPortPeekForEdge(selection,!alreadySelected)
         }
+    }
 
-        //special case if we are selecting multiple things in a palette
-        if(selectedLocation === Eagle.FileType.Palette){
-            this.hierarchyMode(false)
-            this.rightWindow().mode(Eagle.RightWindowMode.Inspector) 
-            return
-        }
+    getObjectInspectorVisibility : ko.PureComputed<boolean> = ko.pureComputed(() => {
+        const eagle = Eagle.getInstance()
+        return eagle.selectedNode() !== null || eagle.selectedEdge() !== null || eagle.selectedObjects().length > 1
+    }, this);
+
+    getInspectorCollapseBtnState : ko.PureComputed<boolean> = ko.pureComputed(() => {
```

Comment:
Might need some clarification.

What is the InspectorCollapseBtn? What states can it have? The return type is boolean, so maybe this is actually the button's visibility?

Also, why does this function addClass() then removeClass() after a timeout? That is unexpected behaviour for a function that gets state.